File: api/api/Models/School.cs
namespace api.Models;

public class School 
{
    public string? Id { get; set; }
    public string? PosterId { get; set; }
    public SchoolType SchoolType { get; set; }
    public string? SchoolName { get; set; }
    public string? PhoneNumber { get; set; }
    public string? Address { get; set; }
    public string? City { get; set; }
    public string? PostalCode { get; set; }
    public Region Region { get; set; }
}

public enum SchoolType
{
    Primary,
    Secondary
}

File: ui/src/models/school.ts
import { Region } from "./user";

export interface School {
    id: string,
    name: string,
    schoolType: SchoolType,
    phoneNumber: string,
    address: string,
    city: string,
    postalCode: string,
    region: Region
}

export enum SchoolType {
    Primary,
    Secondary
}

export enum PrimarySchoolSubject {
    FrenchImmersion,
    CoreFrench,
    ArtsEd,
    General
}

export enum SecondarySchoolSubject // TODO: ADD ALL 
{
    English,
    Math,
    Science,
    History,
    SocialStudies
}

export enum Grade {
    PreK,
    Kindergarten,
    One,
    Two,
    Three,
    Four,
    Five,
    Six,
    Seven,
    Eight,
    Nine,
    Ten,
    Eleven,
    Twelve
}